core/src/main/scala/org/bykn/bosatsu/Expr.scala

```diff
@@ -177,5 +177,27 @@ object Expr {
         buildLambda(NonEmptyList.of(arg), body1, outer)
     }
 
+  def buildPatternLambda[A](
```

Comment:
Looks like `buildLambda` is only used by `DefStatement` now. So not quite ready to delete it... but soon maybe?

Reply:
this is interesting. Should we allow patterns in defs?
```
def getBar(Bar(x)): x
```
doesn't look much like python, but if you want to go nuts, Haskell has something like this that allows multiple branches:
```
enum Either: Left(a), Right(b)

def map(Right(a), fn): fn(a)
def map(left @ Left(_), _): left
```
I'm not crazy about that syntax. I think a match would be better.

Reply:
My first thought when I see this is someone could get the impression that bosatsu has function overloading. And I think a match is always more clear

But for the non enum case this seems really natural and would be confusing if it were absent I think

Reply:
Consistency is a powerful argument. Okay. I think I agree with your take: yes to patterns in defs, no to multi defs for enums. So harmony with lambdas.